In [10]:
import numpy as np
import pandas as pd
import  scipy.stats                     as  stats
import  statsmodels.stats.proportion    as  smp
import matplotlib.pyplot     as     plt
import seaborn as sns
import statsmodels.api         as     sm
from   statsmodels.formula.api import ols

In [11]:
data=pd.read_csv('HotelTariff.csv')
data.head()

hotel       mpdrev       bpdrev
0      1  3685.879567  4071.758975
1      2  3637.186616  3756.839042
2      3  3525.172562  3912.271186
3      4  3460.323749  4163.903728
4      5  3392.520371  4507.828300

In [12]:
data.tail()

hotel       mpdrev  bpdrev
95     96  3658.493994     NaN
96     97  3518.229792     NaN
97     98  3634.995771     NaN
98     99  3584.856974     NaN
99    100  3670.800815     NaN

In [13]:
data.dropna(subset=['bpdrev'],inplace=True)

In [14]:
mpdrev=np.array(data['mpdrev'])
bpdrev=np.array(data['bpdrev'])
#bpdrev

# Two sample t test for unpaired data 

# Step 1: Define null and alternative hypothesis

In testing whether Mumbai is more expensive than Bangalore so the null hypothesis means mumbai <= banglor(M<=B) per deluxe room expense and alternate hypothesis means mumbai is greater expense than bangalore(M>B) 

* $H_0$: $\mu{M}$ <= $\mu{B}$      
* $H_A$: $\mu{M}$ > $\mu{B}$

# Step 2: Decide the significance level

Here we select 

α= 0.05 and population standard deviation is not known.

# Step 3: Identify the test statistic

* We have two samples and we do not know the population standard deviation.
* Sample sizes for both samples are not same.
* So we use the t distribution and the $t_STAT$ test statistic for two sample unpaired test.

# Step 4: Calculate the p - value and test statistic

** We use the scipy.stats.ttest_ind to calculate the t-test for the means of TWO INDEPENDENT samples of expense of hotel room given the two sample observations. This function returns t statistic and two-tailed p value.**

** This is a two-sided test for the null hypothesis that 2 independent samples have identical average (expected) values. This test assumes that the populations have identical variances.**

In [15]:
t_statistic,p_value = stats.ttest_ind(mpdrev,bpdrev)
print('pvalue is %1.9f'%p_value)
print('tstats is %1.3f'%t_statistic)

pvalue is 0.000000263
tstats is -5.826


# Step 5: Decide to reject or accept null hypothesis

In this ,pvalue is 0.000000263 which is lesser than 5% level of significance

so the statistical decision is to reject the null hypothesis at 5% level of significance

# so there is sufficient evidence to reject the null hypothesis